image/image.go

```diff
@@ -140,7 +140,10 @@ func NewChildImage(img *Image, child ChildConfig, platform string) *Image {
 	rootFS := img.RootFS
 	if rootFS == nil {
 		rootFS = NewRootFS()
+	} else {
+		rootFS = img.RootFS.Clone()
 	}
+
```

Comment:
The initial vallue of `rootFS` seems unused (`rootfs := img.RootFS`)

Probably the following code would be better
```
var rootFS <type>
if img.RootFS == nil {
  rootFS = NewRootFS()
} else {
  rootFS = img.RootFS.Clone()
}
```